### ISSUE TITLE
New task work line was created twice

### ISSUE THREAD

**Issue by dreispt** [odoo/odoo#7391]
Occasionally users report that Task Work lines are created twice: the user adds a line, saves the Task and finds that two lines were saved. 

I was not able to reproduce the issue, but here is an example:
![snap 2015-07-01 at 09 55 15](https://cloud.githubusercontent.com/assets/1246629/8451003/a58ab03c-1fd7-11e5-9142-af939dc63195.png)

Checking on the database records I can see that the creation timestamp of theduplicate lines is 5 seconds apart:
![snap 2015-07-01 at 09 52 12](https://cloud.githubusercontent.com/assets/1246629/8451132/a55dd390-1fd8-11e5-9652-4f3f68cbe816.png)

I'm clueless of what might be the cause, and I'm curious if someone else was bitten by the same issue. 
Any help is appreciated.

**Comment by fclementic2c** [comment]
I noticed that on invoice lines also when I use analytical accounts, modify and click update taxes and attendances also.

But I can't reproduce

**Comment by clonedagain** [comment]
Smells like a deadlock/retry bug, some module does a cr.commit()?

**Comment by dreispt** [comment]
I'm not using any "manual" commits. Could it be related to `mail.thread`?

**Comment by odony** [comment]
Could have been related/similar to #5629 (perhaps that's what @fclementic2c refers to), except that the lines should have the same `create_date` values if they're ultimately created during the same transaction, and the behavior should appear only after getting an initial validation error...

**Comment by wtaferner** [comment]
Could it be webclient related? I would assume from what is described here that there might be a double request or data duplication on JS side...but this is just guessing based on my experience what could go wrong related to user interaction and handling :-) Anyway, without clear reproduction it is always difficult to identify.

**Comment by dreispt** [comment]
@odony I was finally able to reproduce on Runbot (v8.0): 
- edit an existing Task and add a Work line (to have work on tasks available you may need to go to Settings / Projects first)
- When saving the form, repeatedly click on the "Save" button. 

After the first click the Save button is disabled, but at some point it becomes enabled again for a short time and we are able to click on it again. This may happen naturally on "hyperactive" user with a slow computer.

![snap 2015-07-02 at 11 28 29](https://cloud.githubusercontent.com/assets/1246629/8475236/756b501e-20ae-11e5-959a-fe1d04380ff6.png)

**Comment by leahim** [comment]
@dreispt there's a fix for #5629 posted..

**Comment by Yenthe666** [comment]
Hi @dreispt,

Thank you for your bug report.
At this point Odoo 8 is no longer supported as Odoo only supports three versions (9.0, 10.0, 11.0 at the moment).
If you can still reproduce this issue on Odoo 9, Odoo 10 or Odoo 11 we encourage you to create a new bug report.

Regards,
Yenthe
